modules/zivid/camera.py

```diff
@@ -47,9 +47,128 @@ def __str__(self):
+        overlapping field-of-view then you need to take consideration and sequence the capture calls to avoid the
+        captures interfering with each other.
+
+        Args:
+            settings: Settings to use for the capture.
+
+        Returns:
+            A frame containing a 3D point cloud, a 2D color image, and metadata.
+
+        Raises:
+            TypeError: If the settings argument is not a Settings instance.
+        """
+        if not isinstance(settings, Settings):
+            raise TypeError(
+                "Unsupported type for argument settings. Got {}, expected {}.".format(
+                    type(settings), Settings.__name__
+                )
+            )
+        return Frame(self.__impl.capture2D3D(_to_internal_settings(settings)))
+
+    def capture3d(self, settings):
+        """Capture a single 3D frame.
+
+        This method is used to capture a 3D frame without a 2D color image. It ignores all color settings in the input
+        settings. See `capture2d3d` for capturing a 2D+3D frame.
+
+        This method returns right after the acquisition of the images is complete, and the camera has stopped projecting
+        patterns. For more information, see the remarks section of `capture2d3d` above. Those remarks apply for both 2D,
+        3D, and 2D+3D captures.
+
+        Args:
+            settings: Settings to use for the capture.
+
+        Returns:
+            A frame containing a 3D point cloud and metadata.
+
+        Raises:
+            TypeError: If the settings argument is not a Settings
+        """
+        if not isinstance(settings, Settings):
+            raise TypeError(
+                "Unsupported type for argument settings. Got {}, expected {}.".format(
+                    type(settings), Settings.__name__
+                )
+            )
+        return Frame(self.__impl.capture3D(_to_internal_settings(settings)))
+
+    def capture2d(self, settings):
+        """Capture a single 2D frame.
+
+        This method returns right after the acquisition of the images is complete, and the camera has stopped projecting
+        patterns. For more information, see the remarks section of `capture2d3d` above. Those remarks apply for both 2D,
+        3D, and 2D+3D captures.
+
+        This overload is provided for convenience. Note that only the Settings2D part under `Settings.color` will be
+        used for the capture. The other parts of the settings will be ignored.
+
+        This method will throw if `Settings.color` is not set.
+
+        Args:
+            settings: Settings to use for the capture. Can be either a Settings2D instance or a Settings instance.
+
+        Returns:
+            A Frame2D containing a 2D image and metadata
+
+        Raises:
+            TypeError: If the settings argument is not a Settings2D or a Settings
+        """
+        if isinstance(settings, Settings2D):
+            return Frame2D(self.__impl.capture2D(_to_internal_settings2d(settings)))
```

Comment:
why using differen casing (2D) in the impl method?

Reply:
Ah, I'll change it